### PR TITLE
[TT-12823, DX-1604] Update documentation for master

### PR DESCRIPTION
**Pull request by buger**
### **User description**
Triggered by: lghiur

Included:

Tyk Gateway: true
Tyk Dashboard: true
Tyk MDCB false
Tyk Pump false

Intended for: master
Changes sourced from: release-5.5.0
Config info generator branch: main

Note: Docs updates for 5.5.0 (branch suffix: docs)

JIRA: https://tyktech.atlassian.net/browse/TT-12823

___

### **PR Type**
Documentation, Enhancement

___

### **Description**
- Updated API documentation for Tyk Dashboard and Gateway to version 5.5.0, including new endpoints and schema definitions.
- Corrected spelling errors and updated descriptions in configuration markdown files.
- Enhanced and clarified Tyk Gateway configuration documentation with additional fields and better organization.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>dashboard-swagger.yml</strong><dd><code>Update Dashboard API documentation to version 5.5.0</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/assets/others/dashboard-swagger.yml

<li>Updated API version from 5.4.0 to 5.5.0.<br> <li> Added new endpoints for managing certificates.<br> <li> Introduced new schema definitions for rate limiting and certificate <br>dependencies.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5187/files#diff-97d3961fa92a4034f55bb3ca902c734bec0a55b4f9b4fc6466f6750da7ca5021">+276/-2</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gateway-swagger.yml</strong><dd><code>Update Gateway API documentation to version 5.5.0</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/assets/others/gateway-swagger.yml

<li>Updated API version from 5.4.0 to 5.5.0.<br> <li> Added new schema definitions for rate limiting.<br> <li> Introduced <code>is_ca</code> field in certificate schemas.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5187/files#diff-86641d5f43d842a51680c692c0ca3cd7d7e67776df7bf1409cf97f83a885ad99">+34/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>dashboard-config.md</strong><dd><code>Correct spelling and update port whitelist description</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/shared/dashboard-config.md

<li>Corrected spelling from "organisation" to "organization".<br> <li> Updated description for <code>ports_whitelist</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5187/files#diff-2af548b55c5ac98b8a1c1c3f8e44cb38709a34e988c684247b81029f2b173433">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gateway-config.md</strong><dd><code>Correct spelling and update port whitelist description</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/shared/gateway-config.md

<li>Corrected spelling from "neighbour" to "neighbor".<br> <li> Updated description for <code>ports_whitelist</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5187/files#diff-0a53845669feaeb0d62d4b17d24bfee5fc23482ec7562bd52e23db03373f14d0">+6/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>x-tyk-gateway.md</strong><dd><code>Enhance and clarify Tyk Gateway configuration documentation</code></dd></summary>
<hr>

tyk-docs/content/shared/x-tyk-gateway.md

<li>Added new fields and descriptions for various configurations.<br> <li> Corrected and clarified existing field descriptions.<br> <li> Reorganized sections for better readability.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5187/files#diff-c4c21855fe9a7bb5c25d2f53fd19b2c5afec3ef3f8ca54c0105d8bfe197ff31c">+181/-149</a></td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/assets/others/dashboard-swagger.yml [hunk 1]
```diff
@@ -3858,6 +4061,28 @@ components:
           x-go-name: Events
       type: object
       x-go-package: github.com/TykTechnologies/tyk/apidef
+    RateLimitMeta:
```
Comment (dcs3spp) [inline tyk-docs/assets/others/dashboard-swagger.yml:4064]: Where is RateLimitMeta linked from? should there be an endpoint in this PR or should this be linked to from an existing endpoint?

File: tyk-docs/content/shared/dashboard-config.md [hunk 2]
```diff
@@ -324,7 +324,7 @@ auto configure based on currently MySQL version
 ENV: <b>TYK_DB_ADMINSECRET</b><br />
 Type: `string`<br />
 
-This secret is to be used by a special set of endpoints that we call “Admin APIs”. This API is part of the super-admin context and therefore has a separate endpoint prefix `/admin`. It also requires a special auth header called admin-auth. This purpose of these endpoints is to allow functionality that regular Dashboard users should not have, such as create new organisations, create super users etc. See the [Admin API](https://tyk.io/docs/dashboard-admin-api/) for more information on these endpoints.
+This secret is to be used by a special set of endpoints that we call “Admin APIs”. This API is part of the super-admin context and therefore has a separate endpoint prefix `/admin`. It also requires a special auth header called admin-auth. This purpose of these endpoints is to allow functionality that regular Dashboard users should not have, such as create new organizations, create super users etc. See the [Admin API](https://tyk.io/docs/dashboard-admin-api/) for more information on these endpoints.
```
Comment (dcs3spp) [inline tyk-docs/content/shared/dashboard-config.md:327]:
```suggestion
This secret is to be used by a special set of endpoints that we call “Admin APIs”. This API is part of the super-admin context and therefore has a separate endpoint prefix `/admin`. It also requires a special auth header called admin-auth. The purpose of these endpoints is to allow functionality that regular Dashboard users should not have, such as create new organizations, create super users etc. See the [Admin API](https://tyk.io/docs/dashboard-admin-api/) for more information on these endpoints.
```

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 3]
```diff
@@ -1885,23 +1929,14 @@ The following values are valid:
...
 
+The default `url` is used as the input source.
+
 ### **URLRewriteRule**
 
 URLRewriteRule represents a rewrite matching rules.
 
 **Field: `in` ([URLRewriteInput](#urlrewriteinput))**
 In specifies one of the valid inputs for URL rewriting.
-By default, it uses `url` as the input source.
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1894]: Why have details concerning the default and valid values been removed? Is it better to include them?

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 4]
```diff
@@ -1925,10 +1960,7 @@ such that the rewrite will be triggered if the value does not match the `pattern
 URLRewriteTrigger represents a set of matching rules for a rewrite.
 
 **Field: `condition` ([URLRewriteCondition](#urlrewritecondition))**
-Condition indicates the logical combination that will be applied to the rules for an advanced trigger:
-
-- Value `any` means any of the defined trigger rules may match
-- Value `all` means all the defined trigger rules must match
+Condition indicates the logical combination that will be applied to the rules for an advanced trigger.
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1963]: Why have valid values been removed, is it better to include them?

**Review by dcs3spp — CHANGES_REQUESTED** [review]
@lghiur Thanks for raising the 5.5 PR @lghiur, PR LGTM! Have raised minor suggestions/questions/observations. Can a peer review also be raised to check expected items present etc?

File: tyk-docs/content/shared/gateway-config.md [hunk 5]
```diff
@@ -363,9 +363,11 @@ In this configuration, Tyk Gateway will allow policy management through the Gate
 ENV: <b>TYK_GW_PORTWHITELIST</b><br />
 Type: `PortsWhiteList`<br />
 
-Defines the ports that will be available for the API services to bind to in the following format: `"{“":“”}"`. Remember to escape JSON strings.
-This is a map of protocol to PortWhiteList. This allows per protocol
-configurations.
+Defines the ports that will be available for the API services to bind to in the format
```
Comment (dcs3spp) [inline tyk-docs/content/shared/gateway-config.md:366]: @lghiur The patch should also be present on 5.3.3 released https://tyk.io/docs/5.3/tyk-oss-gateway/configuration/#ports_whitelist

https://tyk.io/docs/5.3/tyk-oss-gateway/configuration/#ports_whitelist

Comment (lghiur) [reply tyk-docs/content/shared/gateway-config.md:366]: @dcs3spp 5.3.3 is out, I propose to update this with 5.3.4

**Review by dcs3spp — APPROVED** [review]

**Pull request merged**